In [1]:
import pandas as pd
import os
import shutil

from tqdm import tqdm

##### 데이터 불러오기

In [2]:
pokemon_df = pd.read_csv('pokemon_dataset.csv')

##### 포켓몬 타입별 학습데이터 생성

In [3]:
unique_type_list = []
unique_type_list += list(pokemon_df['base_type'].unique())
unique_type_list += list(pokemon_df['sub_type'].unique())

In [4]:
unique_type_list = [x for x in unique_type_list if pd.isnull(x) == False and x != 'nan']
unique_type_list = list(set(unique_type_list))

In [7]:
for type in tqdm(unique_type_list) :
    path = f'train_data/{type}'
    if not os.path.exists(path) : os.mkdir(path)
    
    # 첫번째 타입
    tmp_df = pokemon_df[pokemon_df['base_type'] == type]
    for pokemon in list(tmp_df['name']) :
        shutil.copyfile(f'data/{pokemon}.png', f'train_data/{type}/{pokemon}.png')
    
    # 두번째 타입
    tmp_df = pokemon_df[pokemon_df['sub_type'] == type]
    for pokemon in list(tmp_df['name']) :
        shutil.copyfile(f'data/{pokemon}.png', f'train_data/{type}/{pokemon}.png')

100%|██████████| 18/18 [00:50<00:00,  2.79s/it]


##### 데이터 불러오기

In [8]:
kaggle_pokemon_df = pd.read_csv('All_Pokemon.csv',usecols=['Name','Type 1','Type 2'])
kaggle_pokemon_df = kaggle_pokemon_df[kaggle_pokemon_df['Name'].isin(os.listdir('dataset'))]

##### 포켓몬 타입별 학습데이터 생성

In [9]:
unique_type_list = []
unique_type_list += list(kaggle_pokemon_df['Type 1'].unique())
unique_type_list += list(kaggle_pokemon_df['Type 2'].unique())

In [10]:
unique_type_list = [x for x in unique_type_list if pd.isnull(x) == False and x != 'nan']
unique_type_list = list(set(unique_type_list))

In [11]:
for type in tqdm(unique_type_list) :
    path = f'train_data/{type}'
    if not os.path.exists(path) : os.mkdir(path)
    
    # 첫번째 타입
    tmp_df = kaggle_pokemon_df[kaggle_pokemon_df['Type 1'] == type]
    for pokemon in list(tmp_df['Name']) :
        for img_file in [file for file in os.listdir(f'dataset/{pokemon}') if file.endswith('.jpg') | file.endswith('.png')] :
            shutil.copyfile(f'dataset/{pokemon}/{img_file}', f'train_data/{type}/{img_file}')
    
    # 두번째 타입
    tmp_df = kaggle_pokemon_df[kaggle_pokemon_df['Type 2'] == type]
    for pokemon in list(tmp_df['Name']) :
        for img_file in [file for file in os.listdir(f'dataset/{pokemon}') if file.endswith('.jpg') | file.endswith('.png')] :
            shutil.copyfile(f'dataset/{pokemon}/{img_file}', f'train_data/{type}/{img_file}')

100%|██████████| 17/17 [04:13<00:00, 14.92s/it]


##### 학습 데이터셋 사이즈 확인

In [12]:
data_size = 0
for folder in os.listdir('train_data') :
    size = len(os.listdir(f'train_data/{folder}'))
    print(f'{folder} : {size}')
    data_size += len(os.listdir(f'train_data/{folder}'))

print(f'Total data size : {data_size}')

강철 : 220
격투 : 579
고스트 : 240
노말 : 1627
독 : 2172
드래곤 : 309
땅 : 964
물 : 2482
바위 : 774
벌레 : 811
불꽃 : 1095
비행 : 1457
악 : 73
얼음 : 395
에스퍼 : 1193
전기 : 866
페어리 : 362
풀 : 1176
Total data size : 16795


강철 : 234
격투 : 582
고스트 : 240
노말 : 1634
독 : 2182
드래곤 : 311
땅 : 967
물 : 2499
바위 : 778
벌레 : 818
불꽃 : 1099
비행 : 1463
악 : 73
얼음 : 411
에스퍼 : 1211
전기 : 875
페어리 : 366
풀 : 1182
Total data size : 16925